spython/main/parse/environment.py

```diff
@@ -36,14 +36,13 @@ def parse_env(envlist):
 
     exports = [] 
 
-    # This will need a lot of love!
-    # Detail work not yet done here, see link above for possible cases
-
     for env in envlist:
 
-        pieces = re.split("( |\\\".*?\\\"|'.*?')", env)
-        pieces = [p for p in pieces if p.strip()]
-        export = "=".join(pieces)
-        exports.append(export)
+        if "=" in env:
+            exports = exports + re.split(' ', env)
```

Comment:
will this handle the case `ENV A=B X="Y Y Y"`?

Reply:
I don't think that is valid, at least per their docs. If you use the second format, the equals must be present I think? See https://docs.docker.com/engine/reference/builder/#env


> The second form, ENV <key>=<value> ..., allows for multiple variables to be set at one time. Notice that the second form uses the equals sign (=) in the syntax, while the first form does not. Like command line parsing, quotes and backslashes can be used to include spaces within values.

Reply:
see my function below! It's not pretty, but possibly would meet all cases.